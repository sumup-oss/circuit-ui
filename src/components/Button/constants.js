import PropTypes from 'prop-types';

import { sizes } from '../../styles/constants';

const { KILO, MEGA, GIGA } = sizes;

export const SIZE_PROP_TYPE = PropTypes.oneOf([KILO, MEGA, GIGA]);

export const BUTTON_PROP_TYPES = {
  /**
   * URL the Button should lead to. Causes the Button to render an <a> tag.
   */
  href: PropTypes.string,
  /**
   * Should the Button be disabled?
   */
  disabled: PropTypes.bool,
  /**
   * Button has a 'flat' variation, triggered with this prop.
   */
  flat: PropTypes.bool,
  /**
   * Renders a secondary button. Secondary buttons look the same for
   * primary (default) and flat buttons.
   */
  secondary: PropTypes.bool,
  /**
   * Renders a primary button using the brand color.
   */
  primary: PropTypes.bool,
  /**
   * Link target. Should only be passed, if href is passed, too.
   */
  target: PropTypes.string,
  /**
   * Size of the button. Use the Button's KILO, MEGA, or GIGA properties.
   */
  size: SIZE_PROP_TYPE,
  /**
   * Trigger stretch (full width) styles on the component.
   */
  stretch: PropTypes.bool
};

export const BUTTON_DEFAULT_PROPS = {
  disabled: false,
  flat: false,
  size: sizes.MEGA,
  target: null,
  href: null,
  primary: false,
  secondary: false,
  stretch: false
};
