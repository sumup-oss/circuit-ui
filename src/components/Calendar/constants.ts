/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// This file needs be be kept up to date with
// https://github.com/airbnb/react-dates/blob/master/src/constants.js
// react-dates doesn't support tree-shaking. The components can be code-split,
// but these constants are often used as props which makes code-splitting much
// trickier. Moving them into Circuit UI is a simpler solution.

export const DISPLAY_FORMAT = 'L';
export const ISO_FORMAT = 'YYYY-MM-DD';
export const ISO_MONTH_FORMAT = 'YYYY-MM';

export const START_DATE = 'startDate';
export const END_DATE = 'endDate';

export const HORIZONTAL_ORIENTATION = 'horizontal';
export const VERTICAL_ORIENTATION = 'vertical';
export const VERTICAL_SCROLLABLE = 'verticalScrollable';

export const NAV_POSITION_BOTTOM = 'navPositionBottom';
export const NAV_POSITION_TOP = 'navPositionTop';

export const ICON_BEFORE_POSITION = 'before';
export const ICON_AFTER_POSITION = 'after';

export const INFO_POSITION_TOP = 'top';
export const INFO_POSITION_BOTTOM = 'bottom';
export const INFO_POSITION_BEFORE = 'before';
export const INFO_POSITION_AFTER = 'after';

export const ANCHOR_LEFT = 'left';
export const ANCHOR_RIGHT = 'right';

export const OPEN_DOWN = 'down';
export const OPEN_UP = 'up';

export const DAY_SIZE = 39;
export const BLOCKED_MODIFIER = 'blocked';
export const WEEKDAYS = [0, 1, 2, 3, 4, 5, 6];

export const FANG_WIDTH_PX = 20;
export const FANG_HEIGHT_PX = 10;
export const DEFAULT_VERTICAL_SPACING = 22;

export const MODIFIER_KEY_NAMES = new Set(['Shift', 'Control', 'Alt', 'Meta']);
