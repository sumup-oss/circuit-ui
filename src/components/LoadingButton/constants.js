export const EXIT_ANIMATION_DURATION = 2000;

export const DEFAULT_LOADING_DELAY = 500;

export const LOADING_STATES = {
  DISABLED: 'disabled',
  ACTIVE: 'active',
  SUCCESS: 'success',
  ERROR: 'error'
};
