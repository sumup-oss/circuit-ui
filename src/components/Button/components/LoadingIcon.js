import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled, { css, keyframes } from 'react-emotion';
import { size as sizeMixin } from 'polished';

import { sizes } from '../../../styles/constants';
import { values } from '../../../util/fp';

import PureSpinner from '../../Spinner';
import SuccessSvg from './icons/success.svg';
import { LOADING_STATES, SIZE_PROP_TYPE } from '../constants';

const { KILO, MEGA, GIGA } = sizes;

const iconEnter = keyframes`
  0% {
    opacity: 1;
    transform: scale3d(0, 0, 0);
  }

  100% {
    opacity: 1;
    transform: scale3d(1, 1, 1);
  }
`;

const iconBaseStyles = css`
  transform: scale3d(0, 0, 0);
  opacity: 0;
  transition: opacity 200ms ease-in-out;
  ${sizeMixin('100%')};
`;

const iconVisibleStyles = ({ visible }) =>
  visible &&
  css`
    animation: ${iconEnter} 200ms ease-in-out;
    animation-fill-mode: forwards;
    animation-iteration-count: 1;
  `;

const SuccessIcon = styled(SuccessSvg)(iconBaseStyles, iconVisibleStyles);

// TODO: use default animation after merging.
const centeredStyles = css`
  display: block;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate3d(-50%, -50%, 0);
`;

const sizeStyles = ({ theme, size }) => {
  const sizeMap = {
    [KILO]: theme.spacings.mega,
    [MEGA]: theme.spacings.giga,
    [GIGA]: theme.spacings.tera
  };

  const sizeValue = sizeMap[size] || sizeMap.GIGA;

  return css`
    label: spinner--${size.toLowerCase()};
    ${sizeMixin(sizeValue)};
  `;
};

const SuccessContainer = styled('div')(centeredStyles, sizeStyles);

const Spinner = styled(PureSpinner)(sizeStyles, centeredStyles);

const Success = ({ size, ...props }) => (
  <SuccessContainer size={size}>
    <SuccessIcon {...props} />
  </SuccessContainer>
);

Success.propTypes = {
  size: PropTypes.oneOf(SIZE_PROP_TYPE)
};

Success.defaultProps = {
  size: GIGA
};

const LoadingIcon = ({ loadingState, size }) => (
  <Fragment>
    <Spinner size={size} active={loadingState === LOADING_STATES.ACTIVE} />
    <Success size={size} visible={loadingState === LOADING_STATES.SUCCESS} />
  </Fragment>
);

LoadingIcon.propTypes = {
  loadingState: PropTypes.oneOf(values(LOADING_STATES)).isRequired,
  size: PropTypes.string.isRequired
};

/**
 * @component
 */
export default LoadingIcon;
