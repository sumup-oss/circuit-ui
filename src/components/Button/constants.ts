/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import PropTypes from 'prop-types';

import { sizes } from '../../styles/constants';

const { KILO, MEGA, GIGA } = sizes;

export const SIZE_PROP_TYPE = PropTypes.oneOf([KILO, MEGA, GIGA]);

// TODO: move this back to the component once we Make LoadingButton a
//       sub-component.
export const BUTTON_PROP_TYPES = {
  /**
   * Should the Button be disabled?
   */
  disabled: PropTypes.bool,
  /**
   * URL the Button should lead to. Causes the Button to render an <a> tag.
   */
  href: PropTypes.string,
  /**
   * Renders a primary button using the brand color.
   */
  primary: PropTypes.bool,
  /**
   * Size of the button. Use the Button's KILO, MEGA, or GIGA properties.
   */
  size: SIZE_PROP_TYPE,
  /**
   * Renders a secondary button.
   */
  secondary: PropTypes.bool,
  /**
   * Trigger stretch (full width) styles on the component.
   */
  stretch: PropTypes.bool,
  /**
   * Link target. Should only be passed, if href is passed, too.
   */
  target: PropTypes.string
};

export const BUTTON_DEFAULT_PROPS = {
  disabled: false,
  href: null,
  primary: false,
  secondary: false,
  size: MEGA,
  stretch: false,
  target: null
};
