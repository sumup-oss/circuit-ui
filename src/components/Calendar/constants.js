import * as constants from 'react-dates/constants';

export default constants;
