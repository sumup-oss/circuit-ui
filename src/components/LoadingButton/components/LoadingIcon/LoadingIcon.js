/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css, keyframes } from '@emotion/core';
import { size as sizeMixin } from 'polished';

import { sizes } from '../../../../styles/constants';

import PureSpinner from '../../../Spinner';
import { ReactComponent as SuccessSvg } from '../../icons/success.svg';
import { ReactComponent as ErrorSvg } from '../../icons/error.svg';
import { DISABLED, ACTIVE, SUCCESS, ERROR } from '../../constants';
import { SIZE_PROP_TYPE } from '../../../Button/constants';

const { KILO, MEGA, GIGA } = sizes;

/**
 * Keyframes
 */

const iconEnter = keyframes`
  0% {
    opacity: 1;
    transform: scale3d(0, 0, 0);
  }

  100% {
    opacity: 1;
    transform: scale3d(1, 1, 1);
  }
`;

/**
 * General purpose style functions.
 */

const centeredStyles = () => css`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
`;

const sizeStyles = label => ({ theme, size }) => {
  const sizeMap = {
    [KILO]: theme.spacings.mega,
    [MEGA]: theme.spacings.giga,
    [GIGA]: theme.spacings.tera
  };

  const sizeValue = sizeMap[size] || sizeMap.GIGA;

  return css`
    label: ${`${label}--${size.toLowerCase()}`};
    ${sizeMixin(sizeValue)};
  `;
};

/**
 * Icon styles
 */

const Icon = styled.div`
  label: loading-icon;
  transform: scale3d(0, 0, 0);
  opacity: 0;
  transition: opacity ${({ theme }) => theme.transitions.default};
  ${sizeMixin('100%')};
  animation: ${iconEnter} ${({ theme }) => theme.transitions.default};
  animation-fill-mode: forwards;
  animation-iteration-count: 1;
`;

const IconContainer = styled.div(
  sizeStyles('loading-icon__status'),
  centeredStyles
);

/**
 * Direct sub-components
 */

const Spinner = styled(PureSpinner)(
  sizeStyles('loading-icon__spinner'),
  centeredStyles
);

// TODO: add ARIA labels to icon.
const StatusIcon = ({ as, size }) => (
  <IconContainer size={size}>
    <Icon as={as} />
  </IconContainer>
);

StatusIcon.propTypes = {
  /**
   * Size prop from the Button.
   */
  as: PropTypes.oneOf([SuccessSvg, ErrorSvg]),
  size: SIZE_PROP_TYPE
};

StatusIcon.defaultProps = {
  as: SuccessSvg,
  size: GIGA
};

/**
 * The components center themselves in the relatively positioned parent.
 */
const LoadingIcon = ({ loadingState, size }) => (
  <Fragment>
    <Spinner size={size} active={loadingState === ACTIVE} />
    {loadingState === SUCCESS && <StatusIcon as={SuccessSvg} size={size} />}
    {loadingState === ERROR && <StatusIcon as={ErrorSvg} size={size} />}
  </Fragment>
);

LoadingIcon.propTypes = {
  /**
   * Current loading state of the button. Determines whether the icon shows
   * and whether Spinner or Success are shown.
   */
  loadingState: PropTypes.oneOf([DISABLED, ACTIVE, SUCCESS, ERROR]).isRequired,
  /**
   * Size prop from the Button.
   */
  size: PropTypes.string.isRequired
};

/**
 * @component
 */
export default LoadingIcon;
