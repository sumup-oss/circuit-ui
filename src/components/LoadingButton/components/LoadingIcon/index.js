/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css, keyframes } from '@emotion/core';
import { size as sizeMixin } from 'polished';

import { sizes } from '../../../../styles/constants';
import { values } from '../../../../util/fp';

import PureSpinner from '../../../Spinner';
import { ReactComponent as SuccessSvg } from '../icons/success.svg';
import { LOADING_STATES } from '../../constants';
import { SIZE_PROP_TYPE } from '../../../Button/constants';

const { KILO, MEGA, GIGA } = sizes;

/**
 * Keyframes
 */

const iconEnter = keyframes`
  0% {
    opacity: 1;
    transform: scale3d(0, 0, 0);
  }

  100% {
    opacity: 1;
    transform: scale3d(1, 1, 1);
  }
`;

/**
 * General purpose style functions.
 */

const centeredStyles = () => css`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
`;

const sizeStyles = label => ({ theme, size }) => {
  const sizeMap = {
    [KILO]: theme.spacings.mega,
    [MEGA]: theme.spacings.giga,
    [GIGA]: theme.spacings.tera
  };

  const sizeValue = sizeMap[size] || sizeMap.GIGA;

  return css`
    label: ${`${label}--${size.toLowerCase()}`};
    ${sizeMixin(sizeValue)};
  `;
};

/**
 * Icon styles for success icon
 */

const successIconBaseStyles = ({ theme }) => css`
  label: loading-icon__success;
  transform: scale3d(0, 0, 0);
  opacity: 0;
  transition: opacity ${theme.transitions.default};
  ${sizeMixin('100%')};
`;

const successIconVisibleStyles = ({ theme, visible }) =>
  visible &&
  css`
    label: loading-icon__success--visible;
    animation: ${iconEnter} ${theme.transitions.default};
    animation-fill-mode: forwards;
    animation-iteration-count: 1;
  `;

const SuccessIcon = styled(SuccessSvg)(
  successIconBaseStyles,
  successIconVisibleStyles
);
const SuccessContainer = styled('div')(
  centeredStyles,
  sizeStyles('loading-icon__success')
);

/**
 * Direct sub-components
 */

const Spinner = styled(PureSpinner)(
  sizeStyles('loading-icon__spinner'),
  centeredStyles
);

// TODO: add ARIA labels to icon.
const Success = ({ size, visible }) => (
  <SuccessContainer size={size}>
    <SuccessIcon visible={visible ? 1 : 0} />
  </SuccessContainer>
);

Success.propTypes = {
  /**
   * Size prop from the Button.
   */
  size: SIZE_PROP_TYPE,
  visible: PropTypes.bool
};

Success.defaultProps = {
  size: GIGA,
  visible: false
};

/**
 * Two components that center themselves in the relatively positioned
 * parent.
 */
const LoadingIcon = ({ loadingState, size }) => (
  <Fragment>
    <Spinner size={size} active={loadingState === LOADING_STATES.ACTIVE} />
    <Success size={size} visible={loadingState === LOADING_STATES.SUCCESS} />
  </Fragment>
);

LoadingIcon.propTypes = {
  /**
   * Current loading state of the button. Determines whether the icon shows
   * and whether Spinner or Success are shown.
   */
  loadingState: PropTypes.oneOf(values(LOADING_STATES)).isRequired,
  /**
   * Size prop from the Button.
   */
  size: PropTypes.string.isRequired
};

/**
 * @component
 */
export default LoadingIcon;
