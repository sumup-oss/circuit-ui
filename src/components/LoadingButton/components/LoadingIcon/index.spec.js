import React from 'react';

import { LOADING_STATES } from '../../constants';
import { sizes } from '../../../../styles/constants';
import LoadingIcon from '../LoadingIcon';

const { KILO, MEGA, GIGA } = sizes;

describe('LoadingIcon', () => {
  describe('Style tests', () => {
    it('should have disabled LoadingIcon styles', () => {
      const actual = create(
        <LoadingIcon loadingState={LOADING_STATES.DISABLED} size={KILO} />
      );
      expect(actual).toMatchSnapshot();
    });

    it('should have active LoadingIcon styles', () => {
      const actual = create(
        <LoadingIcon loadingState={LOADING_STATES.ACTIVE} size={KILO} />
      );
      expect(actual).toMatchSnapshot();
    });

    it('should have success LoadingIcon styles', () => {
      const actual = create(
        <LoadingIcon loadingState={LOADING_STATES.SUCCESS} size={KILO} />
      );
      expect(actual).toMatchSnapshot();
    });

    it('should have kilo LoadingIcon styles', () => {
      const actual = create(
        <LoadingIcon loadingState={LOADING_STATES.DISABLED} size={KILO} />
      );
      expect(actual).toMatchSnapshot();
    });

    it('should have mega LoadingIcon styles', () => {
      const actual = create(
        <LoadingIcon loadingState={LOADING_STATES.DISABLED} size={MEGA} />
      );
      expect(actual).toMatchSnapshot();
    });

    it('should have giga LoadingIcon styles', () => {
      const actual = create(
        <LoadingIcon loadingState={LOADING_STATES.DISABLED} size={GIGA} />
      );
      expect(actual).toMatchSnapshot();
    });
  });

  describe('Accessibility tests', () => {
    it('should meet accessibility guidelines', async () => {
      const wrapper = renderToHtml(
        <LoadingIcon loadingState={LOADING_STATES.DISABLED} size={KILO} />
      );
      const actual = await axe(wrapper);
      expect(actual).toHaveNoViolations();
    });
  });
});
