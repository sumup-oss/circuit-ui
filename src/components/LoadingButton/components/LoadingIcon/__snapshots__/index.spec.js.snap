// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIcon Style tests should have active LoadingIcon styles 1`] = `
Array [
  .circuit-1 {
  opacity: 0;
  position: relative;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  opacity: 1;
  height: 16px;
  width: 16px;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<div
    aria-busy="true"
    aria-live="assertive"
    className="circuit-0 circuit-1 circuit-2"
    role="alertdialog"
    size="kilo"
  >
    <div>
      spinner.svg
    </div>
  </div>,
  .circuit-0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 16px;
  width: 16px;
}

<div
    className="circuit-0 circuit-1"
    size="kilo"
  >
    <div>
      success.svg
    </div>
  </div>,
]
`;

exports[`LoadingIcon Style tests should have disabled LoadingIcon styles 1`] = `
Array [
  .circuit-1 {
  opacity: 0;
  position: relative;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  height: 16px;
  width: 16px;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<div
    aria-busy="true"
    aria-live="assertive"
    className="circuit-0 circuit-1 circuit-2"
    role="alertdialog"
    size="kilo"
  >
    <div>
      spinner.svg
    </div>
  </div>,
  .circuit-0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 16px;
  width: 16px;
}

<div
    className="circuit-0 circuit-1"
    size="kilo"
  >
    <div>
      success.svg
    </div>
  </div>,
]
`;

exports[`LoadingIcon Style tests should have giga LoadingIcon styles 1`] = `
Array [
  .circuit-1 {
  opacity: 0;
  position: relative;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  height: 32px;
  width: 32px;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<div
    aria-busy="true"
    aria-live="assertive"
    className="circuit-0 circuit-1 circuit-2"
    role="alertdialog"
    size="giga"
  >
    <div>
      spinner.svg
    </div>
  </div>,
  .circuit-0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 32px;
  width: 32px;
}

<div
    className="circuit-0 circuit-1"
    size="giga"
  >
    <div>
      success.svg
    </div>
  </div>,
]
`;

exports[`LoadingIcon Style tests should have kilo LoadingIcon styles 1`] = `
Array [
  .circuit-1 {
  opacity: 0;
  position: relative;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  height: 16px;
  width: 16px;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<div
    aria-busy="true"
    aria-live="assertive"
    className="circuit-0 circuit-1 circuit-2"
    role="alertdialog"
    size="kilo"
  >
    <div>
      spinner.svg
    </div>
  </div>,
  .circuit-0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 16px;
  width: 16px;
}

<div
    className="circuit-0 circuit-1"
    size="kilo"
  >
    <div>
      success.svg
    </div>
  </div>,
]
`;

exports[`LoadingIcon Style tests should have mega LoadingIcon styles 1`] = `
Array [
  .circuit-1 {
  opacity: 0;
  position: relative;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  height: 24px;
  width: 24px;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<div
    aria-busy="true"
    aria-live="assertive"
    className="circuit-0 circuit-1 circuit-2"
    role="alertdialog"
    size="mega"
  >
    <div>
      spinner.svg
    </div>
  </div>,
  .circuit-0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 24px;
  width: 24px;
}

<div
    className="circuit-0 circuit-1"
    size="mega"
  >
    <div>
      success.svg
    </div>
  </div>,
]
`;

exports[`LoadingIcon Style tests should have success LoadingIcon styles 1`] = `
Array [
  .circuit-1 {
  opacity: 0;
  position: relative;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  height: 16px;
  width: 16px;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<div
    aria-busy="true"
    aria-live="assertive"
    className="circuit-0 circuit-1 circuit-2"
    role="alertdialog"
    size="kilo"
  >
    <div>
      spinner.svg
    </div>
  </div>,
  .circuit-0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 16px;
  width: 16px;
}

<div
    className="circuit-0 circuit-1"
    size="kilo"
  >
    <div>
      success.svg
    </div>
  </div>,
]
`;
